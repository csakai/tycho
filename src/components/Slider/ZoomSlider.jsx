import React from 'react';
import Slider from './Slider';

export default class ZoomSlider extends React.Component {
  render() {
    return (
      <div className="slider slider--vertical">
        <Slider
          orientation="vertical"
          step={10}
          value={this.props.value}
          onChange={this.props.onChange}
        />
      </div>
    );
  }
}
