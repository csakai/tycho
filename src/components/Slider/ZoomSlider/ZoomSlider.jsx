import React from 'react';
import Slider from '../Slider';
import PropTypes from 'prop-types';

export default class ZoomSlider extends React.Component {

  static propTypes = {
    value: PropTypes.number,
    onChange: PropTypes.func
  }

  render() {
    return (
      <div className="slider slider--vertical">
        <Slider
          orientation="vertical"
          invert={true}
          step={1}
          min={1}
          max={100}
          value={this.props.value || 100}
          onChange={this.props.onChange}
        />
      </div>
    );
  }
}
