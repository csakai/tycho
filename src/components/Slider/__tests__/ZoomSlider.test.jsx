import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import ZoomSlider from '../ZoomSlider';

describe('Zoom Slider Component', () => {
  let component;

  beforeEach(() => {
    component = shallow(<ZoomSlider
      value={40}
      onChange={jest.fn()}
    />);
  });

  describe('render()', () => {
    it('should successfully render the slider', () => {
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
