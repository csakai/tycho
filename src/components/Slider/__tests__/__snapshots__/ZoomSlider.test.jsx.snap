// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zoom Slider Component render() should successfully render the slider 1`] = `
<div
  className="slider slider--vertical"
>
  <Slider
    onChange={[Function]}
    orientation="vertical"
    step={10}
    value={40}
  />
</div>
`;
